src/state/pools/hooks.ts

```diff
@@ -30,8 +30,8 @@ export const useFetchPublicPoolsData = () => {
   useSlowRefreshEffect(() => {
     const fetchPoolsDataWithFarms = async () => {
       const activeFarms = nonArchivedFarms.filter((farm) => farm.pid !== 0)
-      await dispatch(fetchFarmsPublicDataAsync(activeFarms.map((farm) => farm.pid)))
-      batch(() => {
+      batch(async () => {
+        await dispatch(fetchFarmsPublicDataAsync(activeFarms.map((farm) => farm.pid)))
```

Comment:
I'm not sure if moving `async/await` dispatch could reduce re-render. That would be great if you could prove this code could reduce re-render times.

Reply:
Actually pools page not rely on the data directly that fetchFarmsPublicDataAsync pushes to redux (it will be used inside the fetchPoolsPublicDataAsync), so you are right it will not reduce the render but I thought it would be nice to have it all of them inside the batch.